todo-list/src/DayList.js

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { ACCENT_COLOURS, COLOURS } from "./colours";
+
+const DayList = (props) => {
+  const numTodoItems = 10;
+  const { day, isDarkMode, accentColour } = props;
+  const [isChecked, setChecked] = useState(
+    Array.from({ length: numTodoItems }).fill(false)
+  );
+  const [textInputValues, setTextInputValues] = useState(
+    Array.from({ length: numTodoItems }).fill("")
+  );
+
+  const todoItemStyles = {
+    color: isDarkMode ? COLOURS.White : COLOURS.Black,
+    backgroundColor: isDarkMode ? COLOURS.Black : COLOURS.White,
+  };
+
+  const checkBoxStyles = {
+    accentColor: accentColour,
+  };
+
+  const handleCheckboxChange = (index) => {
+    const newCheckedState = [...isChecked];
+    newCheckedState[index] = !newCheckedState[index];
+    setChecked(newCheckedState);
+  };
+
+  const handleTextInputChange = (event, index) => {
+    const { value } = event.target;
+    setTextInputValues((prevValues) => {
+      const updatedValues = [...prevValues];
+      updatedValues[index] = value;
+      return updatedValues;
+    });
+  };
+
+  const handleKeyDown = (event, index) => {
+    if (event.key === "Enter" && textInputValues[index]) {
+      const nextIndex = index + 1;
+      if (nextIndex < numTodoItems) {
+        document.getElementById(`todo-input-${day}-${nextIndex}`).focus();
+      }
+    }
+    if (event.key === "Backspace" && textInputValues[index] == "") {
+      const nextIndex = index === 0 ? 0 : index - 1;
+      event.preventDefault();
+      document.getElementById(`todo-input-${day}-${nextIndex}`).focus();
+    }
+  };
+
+  return (
+    <div className="day-todo-list">
```

Comment:
This component is starting to get a bit bloated specifically for the jsx/html part. I think it would be good to have a TodoItem component. Also, I think I might need it for the stuff I am working on. You can do it now, or I am happy to do it for the next PR 😊

Reply:
Sure! I won't be able to work on this until tomorrow/monday so you can give it a try!